nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -54,8 +54,6 @@ let
     ));
   in listToAttrs (map mkAuthKeyFile usersWithKeys);
 
-  supportOldHostKeys = !versionAtLeast config.system.stateVersion "15.07";
```

Comment:
In the release notes, after fixing their host keys, they can safely change their state version from anything prior to `17.03`, to `17.03`, as vetted here: https://search.nix.gsc.io/?q=stateVersion